07_coefficient_analysis.ipynb

```diff
@@ -821,6 +822,460 @@
+      "text/plain": [
+       "<Figure size 1152x432 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "sns.set({'figure.figsize': (16, 6)})\n",
+    "fig, axarr = plt.subplots(1, 2)\n",
+    "\n",
+    "sns.histplot(data=inter_cancer_jaccard_df[inter_cancer_jaccard_df.train_set == 'single cancer'],\n",
+    "             x='mean_jaccard', stat='count', hue='reject_null', element='step',\n",
+    "             kde=False, ax=axarr[0])\n",
+    "axarr[0].set_xlabel('Inter-cancer type, same gene Jaccard index')\n",
+    "axarr[0].set_ylabel('Bin count')\n",
+    "axarr[0].set_title('Same gene/different cancer types, single cancer')\n",
+    "sns.histplot(data=inter_cancer_jaccard_df[inter_cancer_jaccard_df.train_set == 'pancancer'],\n",
+    "             x='mean_jaccard', stat='count', hue='reject_null', element='step',\n",
+    "             kde=False, ax=axarr[1])\n",
+    "axarr[1].set_xlabel('Inter-cancer type, same gene Jaccard index')\n",
+    "axarr[1].set_ylabel('Bin count')\n",
+    "axarr[1].set_title('Same gene/different cancer types, pancancer')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that for the pancancer coefficient comparisons, the model overlaps (coefficient Jaccard similarities) are fairly high, and there is a lot of separation between well-performing and poor-performing models (with well-performing models having higher overlap). This is expected since the pan-cancer training set is mostly consistent between folds or between experiments (with the exception of the cancer type under consideration), so it makes sense that the resulting model is consistent as well.\n",
+    "\n",
+    "In the single-cancer case, model overlaps are much lower in general, and there is not nearly as much separation between well-performing and poorly performing models. This is a bit surprising since we're only considering models that predict mutations in the same gene, so we'd expect them to use some of the same predictors, but this doesn't really seem to be the case often. This may be due to the high level of correlation/collinearity in gene expression data (there are multiple choices of predictors that can predict the same labels, since many of the predictors convey redundant information).\n",
+    "\n",
+    "Maybe looking at this from a pathway or network perspective would help to elucidate this; that is, even if the model selects different genes between folds, we'd expect them to be in the same pathway, or to be close together in a gene interaction network."
```

Comment:
Aggregating genes into networks/pathways is a good idea. You could also try some form of dropout or masking on the inputs to see which genes have similar weights vs swing depending on how many of their colinear partner genes drop out

Reply:
This is a good idea! I'm not sure I'll get to this in the near future, but I'll create an issue to remind myself, since I think this would be a neat analysis once we have some results we're more confident in.